File: service/scaladsl/pubsub/client/src/main/scala/com/lightbend/lagom/scaladsl/broker/pubsub/GooglePubsubMetadataKeys.scala
package com.lightbend.lagom.scaladsl.broker.pubsub

import java.time.Instant

import com.lightbend.lagom.scaladsl.api.broker.MetadataKey

/**
  * Metadata keys specific to the Google Pub/Sub broker implementation.
  */
object GooglePubsubMetadataKeys {
  val Id: MetadataKey[String] = MetadataKey("pubsubId")
  val Attributes: MetadataKey[Map[String, String]] = MetadataKey("pubsubAttributes")
  val Timestamp: MetadataKey[Instant] = MetadataKey("pubsubTimestamp")
}

File: service/javadsl/pubsub/client/src/main/java/com/lightbend/lagom/javadsl/broker/pubsub/GooglePubsubMetadataKeys.java
package com.lightbend.lagom.javadsl.broker.pubsub;

import com.lightbend.lagom.javadsl.api.broker.MetadataKey;

import java.time.Instant;
import java.util.Map;

/**
 * Metadata keys specific to the Google Pub/Sub broker implementation.
 */
public final class GooglePubsubMetadataKeys {
    private GooglePubsubMetadataKeys() {
    }

    public static final MetadataKey<String> ID = MetadataKey.named("pubsubId");

    public static final MetadataKey<Map<String, String>> ATTRIBUTES = MetadataKey.named("pubsubAttributes");

    public static final MetadataKey<Instant> TIMESTAMP = MetadataKey.named("pubsubTimestamp");
}
